File: CS/CustomListBoxPropertyEditor.Module.Win/BusinessObjects/BusinessObjects.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Xpo;

namespace CustomListBoxPropertyEditor.Module.Win.Editors {
    [DefaultClassOptions]
    public class Contact : BaseObject {
        private TitleOfCourtesy titleOfCourtesy;
        public Contact(Session session) : base(session) { }
                                                           
        public TitleOfCourtesy TitleOfCourtesy {
            get { return titleOfCourtesy; }
            set { SetPropertyValue("TitleOfCourtesy", ref titleOfCourtesy, value); }
        }
        private string firstName;
        public string FirstName {
            get { return firstName; }
            set { SetPropertyValue("FirstName", ref firstName, value); }
        }
        private string lastName;
        public string LastName {
            get { return lastName; }
            set { SetPropertyValue("LastName", ref lastName, value); }
        }
        private Position position;
        [ImmediatePostDataAttribute]
        public Position Position {
            get { return position; }
            set { SetPropertyValue("Position", ref position, value); }
        }
    }
    public enum TitleOfCourtesy { Dr, Miss, Mr, Mrs, Ms };

    public class Position : BaseObject {

        public Position(Session session) : base(session) { }
        private string title;
        public string Title {
            get { return title; }
            set { SetPropertyValue("Title", ref title, value); }
        }
    }
}

File: VB/CustomListBoxPropertyEditor.Module.Win/BusinessObjects/BusinessObjects.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Xpo

Namespace CustomListBoxPropertyEditor.Module.Win.Editors
    <DefaultClassOptions> _
    Public Class Contact
        Inherits BaseObject


        Private titleOfCourtesy_Renamed As TitleOfCourtesy
        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Public Property TitleOfCourtesy() As TitleOfCourtesy
            Get
                Return titleOfCourtesy_Renamed
            End Get
            Set(ByVal value As TitleOfCourtesy)
                SetPropertyValue("TitleOfCourtesy", titleOfCourtesy_Renamed, value)
            End Set
        End Property

        Private firstName_Renamed As String
        Public Property FirstName() As String
            Get
                Return firstName_Renamed
            End Get
            Set(ByVal value As String)
                SetPropertyValue("FirstName", firstName_Renamed, value)
            End Set
        End Property

        Private lastName_Renamed As String
        Public Property LastName() As String
            Get
                Return lastName_Renamed
            End Get
            Set(ByVal value As String)
                SetPropertyValue("LastName", lastName_Renamed, value)
            End Set
        End Property

        Private position_Renamed As Position
        <ImmediatePostDataAttribute> _
        Public Property Position() As Position
            Get
                Return position_Renamed
            End Get
            Set(ByVal value As Position)
                SetPropertyValue("Position", position_Renamed, value)
            End Set
        End Property
    End Class
    Public Enum TitleOfCourtesy
        Dr
        Miss
        Mr
        Mrs
        Ms
    End Enum

    Public Class Position
        Inherits BaseObject

        Public Sub New(ByVal session As Session)
            MyBase.New(session)
        End Sub

        Private title_Renamed As String
        Public Property Title() As String
            Get
                Return title_Renamed
            End Get
            Set(ByVal value As String)
                SetPropertyValue("Title", title_Renamed, value)
            End Set
        End Property
    End Class
End Namespace

